js/foam/u2/Element.js

```diff
@@ -478,6 +486,15 @@ CLASS({
       return this;
     },
 
+    // Constructs a default class name for this view, with an optional extra.
+    // Without an extra, results in eg. 'foam-u2-Input'.
+    // With an extra of "foo", results in 'foam-u2-Input-foo'.
+    function myCls(opt_extra) {
+      var base = (this.model_.CSS_CLASS || this.model_.id.replace(/\./g, '-'));
```

Comment:
Can you cache the replace, or since we do this replace a lot, move it to a memeoize1()'ed helper method?